File: DataFrom20240701To20241124/T383.cpp
#include <iostream>
#include <string>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int> arr(26);
        for(int i = 0;i < magazine.size();i ++){
            arr[magazine[i] - 'a'] ++;
        }
        for(int i = 0;i < ransomNote.size();i ++){
            arr[ransomNote[i] - 'a'] --;
        }
        for(int i = 0;i < 26;i ++){
            if(arr[i] < 0) return false;
        }
        return true;
    }
};

File: Leetcode/CodeDataStartFrom20250808/T383.java
import java.util.HashMap;
import java.util.Map;

public class T383 {
    public boolean canConstruct(String ransomNote, String magazine) {
        Map<Character, Integer> mag = new HashMap<>();
        Map<Character, Integer> ran = new HashMap<>();
        int m_len = magazine.length(), r_len = ransomNote.length();
        for (int i = 0;i < m_len;i ++) {
            char ch = magazine.charAt(i);
            mag.put(ch, mag.getOrDefault(ch, 0) + 1);
        }
        for (int i = 0;i < r_len;i ++) {
            char ch = ransomNote.charAt(i);
            ran.put(ch, ran.getOrDefault(ch, 0) + 1);
        }
        
        
        for (Map.Entry<Character, Integer> fk : ran.entrySet()) {
            char ch = fk.getKey();
            int val = fk.getValue();
            int p = mag.getOrDefault(ch, 0);
            if(p < val) {
                return false;
            }
        }
        return true;
    }
}
